nexus/mgs-updates/src/rot_updater.rs

```diff
@@ -235,8 +249,97 @@ impl SpComponentUpdateHelperImpl for ReconfiguratorRotUpdater {
                         .await
                 })
                 .await?;
+
+            debug!(log, "waiting for boot info to confirm a successful reset");
+            wait_for_boot_info(
+                log,
+                mgs_clients,
+                update.sp_type,
+                update.slot_id,
+                WAIT_FOR_BOOT_INFO_TIMEOUT,
+            )
+            .await?;
             Ok(())
         }
         .boxed()
     }
 }
+
+/// Poll the RoT asking for its boot information. This confirms that the RoT has
+/// been succesfully reset
+pub async fn wait_for_boot_info(
```

Comment:
This is just all pretty much a mechanical change, I took most of the logic out of `wait_for_stage0_next_image_check`